sdk/key_vault/src/key.rs

```diff
@@ -239,6 +247,278 @@ impl Display for SignatureAlgorithm {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub enum EncryptionAlgorithm {
+    #[serde(rename = "A128CBC")]
+    A128Cbc,
+    #[serde(rename = "A128CBCPAD")]
+    A128CbcPad,
+    #[serde(rename = "A128GCM")]
+    A128Gcm,
+    #[serde(rename = "A192CBC")]
+    A192Cbc,
+    #[serde(rename = "A192CBCPAD")]
+    A192CbcPad,
+    #[serde(rename = "A192GCM")]
+    A192Gcm,
+    #[serde(rename = "A256CBC")]
+    A256Cbc,
+    #[serde(rename = "A256CBCPAD")]
+    A256CbcPad,
+    #[serde(rename = "A256GCM")]
+    A256Gcm,
+    #[serde(rename = "RSA-OAEP")]
+    RsaOaep,
+    #[serde(rename = "RSA-OAEP-256")]
+    RsaOaep256,
+    #[serde(rename = "RSA1_5")]
+    Rsa15,
+}
+
+impl Default for EncryptionAlgorithm {
+    fn default() -> Self {
+        EncryptionAlgorithm::A128Cbc
+    }
+}
+
+impl Display for EncryptionAlgorithm {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DecryptParameters {
+    pub decrypt_parameters_encryption: DecryptParametersEncryption,
+    #[serde(serialize_with = "ser_base64", deserialize_with = "deser_base64")]
+    pub ciphertext: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum DecryptParametersEncryption {
+    Rsa15(Rsa15Parameters),
```

Comment:
I think this could all be simplified to families. RSA will only ever take the ciphertext and algorithm, while AES GCM will take the ciphertext, aad, and tag, and AES CBC the ciphertext and iv. Would it be possible to simplify these? Overall, though, I appreciate the attention to detail since the parameters do take different arguments.

Reply:
I simplified but had to add an error in the process, in order to make sure the params can't be created with the wrong alg. Let me know if that's ok, or even if there is a better way